doc/Data Plane Code Generation/DeveloperDrivenEvolution.md

```diff
@@ -0,0 +1,145 @@
+
+**Create the models for the convenience APIs you've added.** Currently it can be generated manually. We will support generate it automatically.
+
+**Add helper method mapping between raw response and model in generated or handcrafted model.** Examples for model `MetricFeedback` (Models/MetricFeedback/MetricFeedback.cs) are:
+```C#
+namespace Azure.AI.MetricsAdvisor
+{
+    public partial class MetricFeedback
+    {
+        // Mapping raw response to model
+        internal static MetricFeedback FromResponse(Response response)
+        {
+            using var document = JsonDocument.Parse(response.Content);
+            // Add your deserialization logic in DeserializeMetricFeedback
+            return DeserializeMetricFeedback(document.RootElement);
+        }
+
+        // Mapping model to raw response
+        internal static RequestContent ToRequestContent(MetricFeedback metricFeedback)
+        {
+            var content = new Utf8JsonRequestContent();
+            content.JsonWriter.WriteObjectValue(metricFeedback);
+            return content;
+        }
+    }
+}
+```
+**Call the protocol method and map returned raw response to model.** You could call the protocol method inside your convenience method and map the returned raw response to model. See different scenarios and corresponding examples below. 
+
+
+## Improve a GET method that returns raw JSON to return a model
+
+**Generated code before (Generated/MetricsAdvisorClient.cs):**
+``` C#
+namespace Azure.AI.MetricsAdvisor
+{
+    public partial class MetricsAdvisorClient
+    {
+        public virtual async Task<Response> GetMetricFeedbackAsync(Guid feedbackId, RequestContext context = null)
+        {
+        }
+    }
+}
+```
+**Improve the GET method (MetricsAdvisorClient.cs):**
+``` C#
+namespace Azure.AI.MetricsAdvisor
+{
+    public partial class MetricsAdvisorClient
+    {
+        // Suggest appending value to method name
+        public virtual async Task<Response<MetricFeedback>> GetMetricFeedbackValueAsync(Guid feedbackId, CancellationToken cancellationToken = default)
+        {
+            // Call protocol method
+            Response response = await GetMetricFeedbackAsync(feedbackId, new RequestContext() { CancellationToken = cancellationToken });
+
+            // Calling deserialization helper
+            MetricFeedback value = MetricFeedback.FromResponse(response);
+        }
+    }
+}
+```
+
+## Improve a GET paging method that returns raw JSON to return a model
+
+**Generated code before (Generated/MetricsAdvisorClient.cs):**
+``` C#
+namespace Azure.AI.MetricsAdvisor
+{
+    public partial class MetricsAdvisorClient
+    {
+        public virtual AsyncPageable<BinaryData> GetMetricFeedbacksAsync(Guid feedbackId, RequestContext context = null)
+        {
+        }
+    }
+}
+```
+**Improve the GET paging method (MetricsAdvisorClient.cs):**
```

Comment:
Let's move the paging example after the GET and POST examples, since paging is more complicated.  Is this all they'll need to do?

Reply:
Yes, the complicated logic is in PageableHelpers.Select.